### PR TITLE
Allow json api renderer to be used as test request renderer class

### PR DESCRIPTION
**Pull request by sliverc**
This allows to define json api format as default in tests. Following needs to be configured:

```python
REST_FRAMEWORK {
   ...
    'TEST_REQUEST_RENDERER_CLASSES': (
        'rest_framework_json_api.renderers.JSONRenderer',
    ),
    'TEST_REQUEST_DEFAULT_FORMAT': 'vnd.api+json'
   ...
}
```

With this change tests can be simplified and there is no need to specifically assign json api content type and dump resp. load data as json. This PR changes tests accordingly.

More detail documentation on this rest framework feature can be found [here](http://www.django-rest-framework.org/api-guide/testing/#configuration)

### CODE REVIEW

File: rest_framework_json_api/utils.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def get_resource_name(context, expand_polymorphic_types=False):
 
     # Sanity check to make sure we have a view.
     if not view:
-        raise APIException(_('Could not find view.'))
```
Comment (mblayman) [inline rest_framework_json_api/utils.py]: Why was this switched to returning None instead of raising an exception?

File: example/tests/utils.py [hunk 2]
```diff
@@ -1,21 +1,7 @@
 import json
 
-from django.utils.encoding import force_bytes, force_text
+from django.utils.encoding import force_text
 
 
 def load_json(data):
```
Comment (mblayman) [inline example/tests/utils.py]: Many of the code changes removed `load_json`. Is it still used somewhere or could it be deleted too?

Comment (sliverc) [reply example/tests/utils.py]: I would like to remove it as well. Unfortunalty there is one [test](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/example/tests/integration/test_non_paginated_responses.py#L117) which still needs it.
Reason being is that it uses `RequestFactory` and not `APIClient`. In such a case response doesn't seem to provide a `.json()` function.

File: rest_framework_json_api/utils.py (hunk 1 above)
Comment (sliverc) [reply rest_framework_json_api/utils.py]: DRF uses the test renderer to parse data passed on to `APIClient`. In such a case it won't have a renderer context and therefore there is no view provided.
The json api renderer should still not fail in such a case and render data as is. Hence I had to remove this exception.
Interesting enough is though that the json api renderer was already aware of `resource_name` with value `None`.

File: example/tests/utils.py (hunk 2 above)
Comment (mblayman) [reply example/tests/utils.py]: Would it be possible to convert the test to use `APIClient` instead?

File: rest_framework_json_api/utils.py (hunk 1 above)
Comment (mblayman) [reply rest_framework_json_api/utils.py]: ok, if I understand this correctly based on what you wrote, missing a view should not be possible in a non-test setting? Does that sound true?

Comment (sliverc) [reply rest_framework_json_api/utils.py]: yep this is true...

File: example/tests/utils.py (hunk 2 above)
Comment (sliverc) [reply example/tests/utils.py]: Possibly. I have a look.

Comment (sliverc) [reply example/tests/utils.py]: I have found an ok way to convert it to `APIClient`. Now also removed `example/tests/utils.py`.

**Pull request merged**